components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java

```diff
@@ -199,36 +201,40 @@ public static void setBundleContext(BundleContext bundleContext) {
 
     public static void initializeRegistry(int tenantId, String tenantDomain) throws IdentityException {
 
-        if (tenantId != MultitenantConstants.SUPER_TENANT_ID) {
-            try {
-                PrivilegedCarbonContext.startTenantFlow();
-                PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
-                carbonContext.setTenantDomain(tenantDomain, true);
-                BundleContext bundleContext = IdentityTenantUtil.getBundleContext();
-                if (bundleContext != null) {
-                    ServiceTracker tracker = new ServiceTracker(bundleContext, AuthenticationObserver.class.getName(), null);
-                    tracker.open();
-                    Object[] services = tracker.getServices();
-                    if (services != null) {
-                        for (Object service : services) {
-                            ((AuthenticationObserver) service).startedAuthentication(tenantId);
+        if (tenantIdMap.get(tenantId) == null || Boolean.valueOf(!tenantIdMap.get(tenantId))) {
```

Comment:
! should come before Boolean.valueOf()

Reply:
Still when invocations happens concurrently there's a chance to get the tenants loaded. Isn't it

Reply:
Just noted that the hash map includes a boolean as a value. Why a conversion is needed here